src/HealthGPS/kevin_hall_model.cpp

```diff
@@ -81,74 +94,114 @@ void KevinHallModel::update_risk_factors(RuntimeContext &context) {
         if (person.age == 0) {
             initialise_nutrient_intakes(person);
             initialise_energy_intake(person);
-            initialise_weight(person);
         } else {
             update_nutrient_intakes(person);
             update_energy_intake(person);
         }
     }
 
-    // TODO: Compute target body weight.
-    const float target_BW = 100.0;
-
-    // Trial run.
-    double mean_sim_body_weight = 0.0;
-    double mean_adjustment_coefficient = 0.0;
+    // Update: different model for children and adults.
     for (auto &person : context.population()) {
```

Comment:
I don't see any reason why this loop over population could not be merged with the previous one, saving one whole loop and checking if the person is active.

Reply:
Sadly it's  to do with a very strict and fragile ordering of variables... See the 'chicken and egg' issue #181. My brain cries tears at the thought of unwinding this right now, but I agree it should eventually be done.

Reply:
Actually this was because I didn't want big ugly less readable conditional branches in the for loop. I probably could have squashed it together. in any case, the compiler is generally smart enough to reorganise it more efficiently, so it just boils down to readability preference in the end. Alex likes less indentation, for example.

Reply:
You'd *hope* the compiler would do something clever with it, but sadly I'm not sure you can rely on it. This is a slow path anyway though (right?) so performance doesn't really matter.

I think it would actually be more readable to squash the two loops together though as @dalonsoa suggests.

Reply:
This unfortunately changed in a later PR. The ordering of this model is very brittle (#181), especially now with the latest draft PR. I'd prefer to address this then.